### PR TITLE
Schools Delete Button

### LINKED ISSUE #34: School Delete Button
Part of Epic #4

### PR DESCRIPTION
**Pull request by SR33G33**
Closes #34 
Implemented functionality for the delete button on the School's Index Page

image

When the delete button is clicked

image

The delete branch is deployed at: https://proj-organic-sr33g33-dev.dokku-12.cs.ucsb.edu/

### CODE REVIEW

**Review by jason-rodrigues — DISMISSED** [review]
LGTM

**Comment by pconrad** [comment]
The create school button leads to an error message for me on the dokku deployment linked in the PR.

I can't merge this unless/until I can test it.

**Comment by SR33G33** [comment]
redeployed app and seems to be working on my end, I pushed the most updated working version that has edit and delete and create functionality

note: still has the edit page bugs that I pointed out in the edit page PR which will be fixed in a different PR

File: frontend/src/tests/pages/SchoolCreatePage.test.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    });
+
+    test("on submit, makes request to backend", async () => {
+
+        const queryClient = new QueryClient();
+        const school = {
+            abbrev: "ucsb",
+            name: "UC Santa Barbara",
+            termRegex: "[wsmf]\\d\\d",
+            termDescription: "s24",
```
Comment (pconrad) [inline frontend/src/tests/pages/SchoolCreatePage.test.js]:
```suggestion
            termDescription: "quarter",
```

File: src/main/java/edu/ucsb/cs156/organic/controllers/SchoolController.java [hunk 2]
```diff
@@ -111,34 +110,21 @@ public Object deleteSchool(
...
 
-        if (!abbrev.equals(abbrev.toLowerCase())){
+        if (!school.getAbbrev().equals(school.getAbbrev().toLowerCase())){
             throw new IllegalArgumentException("Invalid abbrev format. Abbrev must be all lowercase");
         }
 
+        if (!school.getTermDescription().matches(school.getTermRegex())) {
+            throw new IllegalArgumentException("Invalid termDescription format. It must follow the pattern " + school.getTermRegex());
+        }
+
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/organic/controllers/SchoolController.java]:
```suggestion
```

Please remove this check; it never made any sense.

**Review by pconrad — CHANGES_REQUESTED** [review]
Changes to update understanding of the purpose of the term field.

File: src/main/java/edu/ucsb/cs156/organic/controllers/SchoolController.java (hunk 2 above)
Comment (SR33G33) [reply src/main/java/edu/ucsb/cs156/organic/controllers/SchoolController.java]: fixed this

**Review by pconrad — APPROVED** [review]
LGTM

**Pull request merged**